### PR TITLE
feat(env-providers): one-time per-project trust modal + Sidebar/Dashboard icon parity

### PR DESCRIPTION
**Pull request by jamesbrink**
## What

Two intertwined fixes:

**a)** Workspace status icons drift between Sidebar rows and Dashboard cards (a merged PR shows the PR-status icon in the sidebar but a generic "idle" icon on the card). Extracted a shared `resolveScmPrIcon` helper so the same `{Icon, color, title}` triple drives both surfaces, PR state always wins over idle. (`4fbf5dab`)

**b)** Untrusted `mise.toml` / `.envrc` files surfaced as an ugly raw-stderr toast and made the workspace unusable until the user manually ran `mise trust` / `direnv allow` from a terminal. Replaced with a one-time-per-project trust modal that lets the user Trust (fan-out across every existing worktree + persist `repo_trust=allow` for future workspaces) or Disable-for-this-project per provider, with proper post-action verification, live progress feedback, and proactive triggering from the Settings panel toggle. (`b365f01c` through `01657712`)

## Why

- Inconsistent status icons made "is this workspace done?" ambiguous depending on where you looked.
- The raw-stderr toast leaked Lua tracebacks + ANSI escape codes to the user, gave no actionable affordance, and required leaving Claudette to fix.
- Future workspaces in the same repo kept hitting the same wall — there was nowhere to "remember this decision."
- Toggling a provider back ON from Settings on a still-untrusted repo silently succeeded in the UI while the next agent spawn still couldn't resolve env vars.

### Backend (`src-tauri/src/commands/env.rs`, `src/env_provider/mod.rs`)

- New `workspace_env_trust_needed` Tauri event emitted from `prepare_workspace_environment` whenever resolved sources include a trust-class error. Carries `{workspace_id, repo_id, plugins: [{plugin_name, message, config_path, error_excerpt}]}`. Non-trust failures continue to surface via the `Err` return so existing toast paths stay intact.
- `is_trust_error_str` promoted to a `pub` heuristic in `env_provider::mod`, mirrored in the TypeScript `TRUST_ERROR_MARKERS` constant so backend + frontend agree on what counts as trust class.
- Error-text cleaner pipeline (`clean_trust_error_excerpt` → `strip_lua_wrapper` → `clean_mise` / `clean_direnv` → `basename`) strips dispatcher prefix + mlua wrapper + Lua call-site location + ANSI codes, then produces a one-line human summary plus an absolute config path. `basename` handles both `/` and `\` so Windows paths survive.
- 20+ unit tests covering each cleaner stage and edge cases (Windows paths, UTF-8 truncation boundaries, Lua traceback wrappers).

### Bundled Lua plugins (`plugins/env-mise/init.lua`, `plugins/env-direnv/init.lua`)

- Tightened error surfacing: instead of `error("mise env failed: " .. stderr)` which dragged the full traceback through, both plugins now regex out the trust line (`Config files in [^\n]+ are not trusted` for mise, `direnv: error [^\n]+ is blocked` for direnv) and `error(trust_line)` directly. Falls back to the previous full-stderr error when the regex misses.

### EnvTrustModal (`src/ui/src/components/modals/EnvTrustModal.tsx`)

- One-time-per-project trust prompt. Per-row state machine: `idle → trusting/disabling → trusted/disabled/failed`. Auto-closes 600ms after every row reaches a terminal user-decision state; failed rows hold open so the user can retry.
- Per-row "Show raw error" disclosure for diagnosing wedges.
- `verifyPluginCleared` re-queries `getEnvSources` after the action and inspects whether the plugin still has a trust-class error — the backend now returns `Ok(())` from `prepare_workspace_environment` when only trust errors remain (they route through the event), so relying on the resolve's exit status alone gives green pills on still-blocked workspaces. Codex P2 finding from the b365f01c review.
- `workspace_id` is now `string | null` so the modal can be opened proactively from Settings without an active workspace. When null, the workspace-scoped reload + re-prepare are skipped and verification falls back to the repo target.
- Live progress feedback: button label ticks at 1Hz showing elapsed wall-clock time (`Trusting… 5s` / `1m 23s`), inline phase indicator below the buttons cycles through `Running \`direnv allow\`…` → `Re-resolving environment…` → `Verifying…`, and after 10s elapsed a `Fanning out to every worktree in this project — this can take a moment.` hint kicks in. UAT caught this — direnv allow across 12 worktrees with no progress feedback felt indistinguishable from a hang.
- Settled rows hold the terminal `Trusted` / `Disabled` button label instead of falling back to the default `Trust` / `Disable for project` — kills the brief flash between row settle and the 600ms auto-close.

### Hook (`src/ui/src/hooks/useWorkspaceEnvironmentPreparation.ts`)

- Listens for `workspace_env_trust_needed`, opens `EnvTrustModal` with the payload.
- Suppresses the matching toast (via `looksLikeTrustError`) so the user sees the modal, not both.

### EnvPanel (`src/ui/src/components/settings/sections/EnvPanel.tsx`)

- Routes all trust UI through the shared modal instead of inline ErrorCard + Trust/Disable/Copy buttons. The Settings panel now reads as a clean status list: status dot + name + badge + var count + age + (Resolve…|Settings|Details) button + toggle.
- On toggle-ON, if the resolved row reports a trust-class error, opens `EnvTrustModal` with `workspace_id: null` so the prompt works without an active workspace (UAT requirement: "I should be able to trust regardless of having an active workspace").
- Subscribes to `activeModal` and refreshes `sources` when `EnvTrustModal` closes — `runEnvTrust` / `setEnvProviderEnabled` don't emit `env-cache-invalidated`, so without this hook the row keeps its ERROR badge and clicking `Resolve…` again would re-open the modal off stale state. Codex P2 from the simplification commit review.
- Non-trust errors (broken TOML, flake eval failure) still get an inline `<pre>` disclosure — the modal isn't the right surface for those, and re-running `mise trust` wouldn't help anyway.
- Settings drawer renders manifest fields only; never co-mingled with error UI.

### Shared formatter util (`src/ui/src/components/modals/envTrustFormat.ts`)

- `summarizeError` — mirrors the backend `clean_trust_error_excerpt` chain on the proactive-from-Settings path so the two entry points produce identical text.
- `formatElapsed` — `Ns` / `Nm Ms` compact label for the in-flight button counter. Unit baked in so translation strings don't accidentally double-append `s` (UAT-caught `Trusting… 5ss` bug).

### Docs (`site/src/content/docs/features/per-repo-settings.mdx`)

- Updated the "One-time per-repo trust prompt" section to describe the modal flow and the persisted `repo_trust=allow` decision.

### i18n (`src/ui/src/locales/en/modals.json`)

- `env_trust_*` keys for title / intro / repo label / action labels / progress labels / phase labels / long-running hint / details disclosure / done / cancel.

## Regression coverage

**New tests**
- `envTrustFormat.test.ts` (16 cases) — every regex prefix the cleaner strips, multi-line collapse, 240-char truncation, empty-after-strip safety, formatElapsed boundaries, explicit `Trusting… 5ss` guard.
- `EnvTrustModal.test.ts` additions — `isEnvTrustModalData` with `workspace_id: null` (proactive shape) and rejection cases (missing field, wrong type, numeric).
- `env.rs` backend tests — 20+ cases covering the cleaner pipeline, Windows path basename, UTF-8 truncation boundary, Lua wrapper variants.

**Existing coverage preserved**
- `allRowsResolved` (8 cases) — modal auto-close threshold, failed rows hold open.
- `classifyPostActionError` (6 cases) — Codex P2 regression guard against marking still-blocked rows trusted.
- `plugin_tests.rs` — bundled plugin trust-line regex stays correct for actual mise/direnv stderr formats.

## Manual UAT

8 repos × 14 worktrees, freshly reset (mise + direnv stamps wiped, `repo_trust` DB rows cleared, env-cache evicted):

1. ✅ Opening a workspace with untrusted `mise.toml` + `.envrc` shows the modal, not the toast.
2. ✅ Clicking Trust runs the command across all worktrees, persists `repo_trust=allow`, verifies post-action, flashes Trusted pill, auto-closes.
3. ✅ Future workspaces in the same repo auto-heal (no modal on second open).
4. ✅ Disable-for-project sets the per-repo enable flag false, dispatcher skips the provider.
5. ✅ Toggling a provider back ON from Settings on a still-untrusted repo opens the modal proactively.
6. ✅ Settings panel works without an active workspace (modal verifies against repo target).
7. ✅ Modal text is clean (no `[0m`, no `[string "..."]:N:`, no `export:` prefix).
8. ✅ Trust button shows `Trusting… 5s` ticking; phase line cycles `Running` → `Re-resolving` → `Verifying`; `Fanning out…` hint appears past 10s.
9. ✅ No `Trusting → Trust → close` flash on auto-close.
10. ✅ Row refreshes from ERROR to fresh/cached the moment the modal closes.
11. ✅ Sidebar and Dashboard show the same PR-state icon for merged PRs.

## Codex peer review

Two passes on this branch caught two genuine P2 regressions during development; both fixed in this branch:
- (b365f01c review) Trust button validated nothing — modal marked rows trusted without verifying the resolve actually cleared. Fixed with `verifyPluginCleared` + `classifyPostActionError`.
- (f69578bd review) `EnvPanel` left stale after modal trust — rows kept showing ERROR until manual Reload. Fixed by subscribing to `activeModal` and refreshing on close transition.

## Test summary

- **Rust**: `cargo test -p claudette -p claudette-server -p claudette-cli --all-features` ✅
- **Rust clippy**: `cargo clippy -p claudette -p claudette-server -p claudette-cli --all-targets --all-features` ✅ (zero warnings under `-Dwarnings`)
- **TypeScript**: `bunx tsc -b` ✅
- **ESLint**: `bun run lint` ✅ (zero errors)
- **CSS tokens**: `bun run lint:css` ✅
- **Vitest**: 157 files / 1890 tests pass

## Commit list

- `4fbf5dab` fix(ui): show same PR status icon on Sidebar row and Dashboard card
- `b365f01c` feat(env-providers): one-time per-project trust modal for mise/direnv
- `5f3f9319` fix(env-trust-modal): address codex peer-review findings on b365f01c
- `f69578bd` refactor(env-providers): route trust through the modal; simplify Settings
- `d842f60a` fix(env-panel): refresh after trust modal closes; strip ANSI from error text
- `65f0c170` feat(env-trust-modal): live progress indicator + fix Trust→Trust flash
- `2b297c45` fix(env-trust-modal): drop duplicate 's' from elapsed counter label
- `01657712` test(env-trust-modal): extract formatters + add regression tests

### CODE REVIEW

File: src/ui/src/components/modals/EnvTrustModal.tsx [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+          );
+        })}
+      </div>
+      <div className={shared.actions}>
+        <button className={shared.btn} onClick={closeModal}>
+          {Object.values(rowStates).some(
+            (s) => s.kind === "trusted" || s.kind === "disabled",
+          )
+            ? t("env_trust_done")
+            : tCommon("cancel")}
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/modals/EnvTrustModal.tsx]: Addressed in ec620297. Switched the cancel button to `t("env_trust_cancel")` ("Decide later") so the locale key is used and matches the modal's docstring/i18n contract. Dropped the now-unused `tCommon` import.

File: src/ui/src/components/modals/EnvTrustModal.tsx [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+                  {Date.now() - state.startedAt > 10_000 && (
+                    <p className={styles.hint}>
+                      {t("env_trust_progress_long_running")}
+                    </p>
+                  )}
+                </>
+              ) : (
+                <p className={styles.hint}>
+                  {t("env_trust_trust_hint", { command })}
+                </p>
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/modals/EnvTrustModal.tsx]: Addressed in ec620297. The idle row now renders both hints stacked — `env_trust_trust_hint` for what Trust does, plus `env_trust_disable_hint` so the disable action also has its own explanatory copy. While the row is busy, both hints are still replaced by the live phase indicator (same as before). No unused i18n keys left.

File: site/src/content/docs/features/per-repo-settings.mdx [hunk 3]
```diff
@@ -61,10 +61,13 @@ The most common reason to set a per-repo override is **timeout**: the default ca
...
-- **Disable for this repo** — skips the provider for this repo entirely. Other env-providers still run. You can re-enable from the row's toggle if you change your mind.
+- **Trust** — runs the trust command (`mise trust` / `direnv allow`) in the repo's main checkout AND every existing worktree under it, then persists `repo_trust = "allow"` so any **future** workspace in this repo auto-trusts silently on first resolve via the env-provider plugin's retry path.
+- **Disable for project** — toggles the per-repo `enabled = false` flag so the dispatcher skips the provider entirely for any worktree under this repo. Other env-providers still run. You can re-enable from the row's toggle in Repo Settings → Environment.
+- **Decide later** — dismisses the modal without persisting a choice. The same prompt surfaces again the next time a workspace fails resolve.
+
+The same actions are also available without the modal from the env-panel error card in Repo Settings → Environment, for users who dismissed the prompt or want to revisit their decision.
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply site/src/content/docs/features/per-repo-settings.mdx]: Addressed in ec620297. Rewrote the paragraph to describe the current flow: row-level **Resolve…** button re-opens the same modal, toggling a disabled provider back ON auto-prompts if the underlying config is still untrusted, non-trust errors stay inline via the **Details** disclosure. The old inline error-card sentence is gone.

File: src/ui/src/hooks/useWorkspaceEnvironmentPreparation.ts [hunk 4]
```diff
@@ -24,6 +24,39 @@ interface WorkspaceEnvProgressPayload {
...
+ * mirroring `WorkspaceEnvTrustNeededPayload` in src-tauri/src/commands/
+ * env.rs. Emitted whenever `prepare_workspace_environment` detects an
+ * untrusted mise / direnv config on the worktree; we route it into the
+ * `envTrust` modal so the user gets a one-time per-project prompt
+ * instead of an opaque toast.
+ */
+interface WorkspaceEnvTrustNeededPayload {
+  workspace_id: string;
+  repo_id: string;
+  plugins: Array<{ plugin_name: string; error_excerpt: string }>;
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/hooks/useWorkspaceEnvironmentPreparation.ts]: Addressed in ec620297. Extended `WorkspaceEnvTrustNeededPayload.plugins[]` to include the optional `message` + `config_path` fields the Rust event actually emits (mirrors `TrustNeededEntry` in env.rs). Both are typed as `string | null | undefined` so an older backend whose payload omits them still type-checks — same tolerance the modal's `isEnvTrustModalData` validator already applies at runtime.

File: src/ui/src/components/settings/sections/EnvPanel.tsx [hunk 5]
```diff
@@ -653,25 +693,10 @@ export function EnvPanel({ target }: EnvPanelProps) {
...
-                  running={runningTrust === source.plugin_name}
-                  onRunTrust={() => handleRunTrust(source.plugin_name)}
-                  onDisableForRepo={
-                    repoIdForOverrides
-                      ? () => handleDisableForRepo(source.plugin_name)
-                      : undefined
-                  }
-                />
+              {isOpen && showDetails && (
+                <pre className={styles.envErrorPre}>{source.error}</pre>
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/settings/sections/EnvPanel.tsx]: Addressed in ec620297. Added a small `ErrorDisclosure` helper wrapping the inline `<pre>` with a Copy button backed by the shared `useCopyToClipboard` hook (same affordance the old ErrorCard had). New `.envErrorDisclosure` CSS class keeps the button right-aligned under the pre. Trust-class errors continue to route through `EnvTrustModal`, which has its own raw-error disclosure + Copy in its details panel.
